proposals/NNNN-android-notifications.md

```diff
@@ -0,0 +1,228 @@
+
+#### Conditions
+
+- An SDL enabled app is in the foreground on the phone
+- The app never established to register to the connected Bluetooth device
+- On the phone, there is no router service provided by another app 
+- The app got connected and registered to the head unit
+
+#### Requirements
+
+1. The app should store that information
+2. The router service shows foreground notifications to the user.
+
+#### Comments
+
+This is the end of the "first time" guide storing the information of the user owning an SDL capable device
+
+#### 4. Router service with the app in background for known SDL BT devices
+
+#### Conditions
+
+- An SDL enabled app is not in the foreground on the phone
+- The app established to register to the connected Bluetooth device
+- On the phone, there is no router service provided by another app 
+
+#### Requirements
+
+The app should start the router service as a foreground service including a notification.
+
+#### Comments
+
+As the app knows that the device supports SDL it can start the router service and keep it active/foregrounded
+
+#### 5. The app connects to router service when in background
+
+#### Conditions
+
+- An SDL enabled app is in the foreground on the phone
+- The app never established to register to the connected Bluetooth device
+- On the phone, another app has created the router service
+
+#### Requirements
+
+1. The app's proxy should connect to the router service
+2. The router service should tell the app if the connected BT device is known to support SDL
+
+#### Comments
+
+This behavior can exist if multiple SDL apps are installed on users device.
+
+## Potential downsides
+
+The downside is the reduce experience for the first time use as the phone must be unlocked and the app must be started. However, this proposal reduces the intrusion of SDL foreground services.
+
+## Impact on existing code
+
+Changes to the library can be high. It's likely this to require a major version change as it could change critical areas of multiplexing.
+
+## Alternatives considered
+
+When the user stopped the Android Router service, the android proxy can remember the MAC address of the device. It can maintain the list of MAC addresses for which the user stopped the Android Router Service. This would be a list of blocked devices. On subsequent Bluetooth connections, the Android Router service will not be started for devices available on the blocked list. The SDL router service can provide the list of blocked devices to SDL App. The App can request this device list and re-enable SDL for the specified device. The downside of the proposal is that SDL will have to provide a standard settings screen. This settings screen can be used to see the list of blocked devices and provide the necessary user interface so that the user can enable SDL for blocked devices. Since the Android Router service maintains the blocked device list, the settings will be shared between all SDL apps.
```

Comment:
```suggestion
When the user stops the Android Router Service, the Android proxy can remember the MAC address of the device. It can maintain the list of MAC addresses for which the user stopped the Android Router Service. This would be a list of blocked devices. On subsequent Bluetooth connections, the Android Router Service would not be started for devices available on the blocked list. The SDL router service could provide the list of blocked devices to SDL app. The app could request this device list and re-enable SDL for the specified device. The downside of this alternative is that SDL would have to provide a standard settings screen. This settings screen could be used to see the list of blocked devices and provide the necessary user interface so that the user could enable SDL for blocked devices. Since the Android Router Service maintains the blocked device list, the settings would be shared between all SDL apps.
```